mri/operators/gradient/gradient.py

```diff
@@ -0,0 +1,165 @@
+    """ Gradient Synthesis class.
+    This class defines the grad operators for:
+            (1/2) * sum(||Ft Psi_t alpha - yl||^2_2,l)
+    Attributes
+    ----------
+    data: np.ndarray
+        input 2D data array.
+    fourier_op: instance
+        a Fourier operator instance.
+    linear_op: instance
+        a linear operator instance.
+    verbose: int, default 0
+        Debug verbosity. Prints debug information during initialization if 1.
+    """
+    def __init__(self, data, linear_op, fourier_op, verbose=0, **kwargs):
+        self.fourier_op = fourier_op
+        self.linear_op = linear_op
+        coef = linear_op.op(np.squeeze(np.zeros((linear_op.n_coils,
+                                                 *fourier_op.shape))))
+        self.linear_op_coeffs_shape = coef.shape
+        super(GradSynthesis, self).__init__(data,
+                                            self._op_method,
+                                            self._trans_op_method,
+                                            self.linear_op_coeffs_shape,
+                                            verbose=verbose,
+                                            **kwargs)
+
+    def _op_method(self, data):
+        return self.fourier_op.op(self.linear_op.adj_op(data))
+
+    def _trans_op_method(self, data):
+        return self.linear_op.op(self.fourier_op.adj_op(data))
+
+
+class GradSelfCalibrationAnalysis(GradBaseMRI):
+    """ Gradient Analysis class for parallel MR reconstruction based on the
+    coil sensitivity profile.
+    This class defines the grad operators for:
+            (1/2) * sum(||Ft Sl x - yl||^2_2,l)
+    Attributes
+    ----------
+    data: np.ndarray
+        input observed data array.
+    fourier_op: instance
+        a Fourier operator instance.
+    Smaps: np.ndarray
+        Coil sensitivity profile [L, Nx, Ny, Nz]
+    verbose: int, default 0
+        Debug verbosity. Prints debug information during initialization if 1.
+    """
+    def __init__(self, data, fourier_op, Smaps, verbose=0, **kwargs):
+        self.Smaps = Smaps
+        self.fourier_op = fourier_op
+        super(GradSelfCalibrationAnalysis, self).__init__(
+            data,
+            self._op_method,
+            self._trans_op_method,
+            fourier_op.shape,
+            verbose=verbose,
+            **kwargs)
+
+    def _op_method(self, data):
+        data_per_ch = np.asarray([data * self.Smaps[ch]
+                                  for ch in range(self.Smaps.shape[0])])
+        return self.fourier_op.op(data_per_ch)
+
+    def _trans_op_method(self, coeff):
+        data_per_ch = self.fourier_op.adj_op(coeff)
+        return np.sum(data_per_ch * np.conjugate(self.Smaps), axis=0)
+
+
+class GradSelfCalibrationSynthesis(GradBaseMRI):
+    """ Gradient Synthesis class for parallel MR reconstruction based on the
+    coil sensitivity profile.
+    This class defines the grad operators for:
+            (1/2) * sum(||Ft Psi_t Sl Alpha - yl||^2_2,l)
```

Comment:
Isn't it `(1/2) * sum(||Ft Sl Psi_t Alpha - yl||^2_2,l)`?

Reply:
I think you are right. This was legacy code, will fix it.